lib/clover_deriv_quda.cu

```diff
@@ -164,39 +61,36 @@ namespace quda {
     } else {
       errorQuda("Force order %d not supported", force.Order());
     } // force / oprod order
-
-    qudaDeviceSynchronize();
   }
-#endif // GPU_CLOVER
 
-  void cloverDerivative(
-      cudaGaugeField &force, cudaGaugeField &gauge, cudaGaugeField &oprod, double coeff, QudaParity parity)
-  {
 #ifdef GPU_CLOVER_DIRAC
-  assert(oprod.Geometry() == QUDA_TENSOR_GEOMETRY);
-  assert(force.Geometry() == QUDA_VECTOR_GEOMETRY);
-
-  for (int d=0; d<4; d++) {
-    if (oprod.X()[d] != gauge.X()[d])
-      errorQuda("Incompatible extended dimensions d=%d gauge=%d oprod=%d", d, gauge.X()[d], oprod.X()[d]);
-  }
-
-  int device_parity = (parity == QUDA_EVEN_PARITY) ? 0 : 1;
+  void cloverDerivative(GaugeField &force, GaugeField &gauge, GaugeField &oprod, double coeff, QudaParity parity)
+  {
+    (void)coeff;
```

Comment:
is this here to silence warnings?

Reply:
Yes, but I realize this isn't the most portable way to do this.  I've cleaned this up to be more portable with b4c8f739e.

Note that this double definition of functions, with the second being a dummy variant with no named variables can all be removed when we go C++17.  That'll make a nice cleanup PR post merge.